Modules/_io/textio.c

```diff
@@ -42,52 +50,76 @@ _unsupported(const char *message)
     return NULL;
 }
 
-PyDoc_STRVAR(textiobase_detach_doc,
-    "Separate the underlying buffer from the TextIOBase and return it.\n"
-    "\n"
-    "After the underlying buffer has been detached, the TextIO is in an\n"
-    "unusable state.\n"
-    );
+/*[clinic input]
+_io._TextIOBase.detach
+    cls: defining_class
+    /
+
+Separate the underlying buffer from the TextIOBase and return it.
+
+After the underlying buffer has been detached, the TextIO is in an unusable state.
+[clinic start generated code]*/
 
 static PyObject *
-textiobase_detach(PyObject *self, PyObject *Py_UNUSED(ignored))
+_io__TextIOBase_detach_impl(PyObject *self, PyTypeObject *cls)
+/*[clinic end generated code: output=50915f40c609eaa4 input=987ca3640d0a3776]*/
 {
     return _unsupported("detach");
 }
 
-PyDoc_STRVAR(textiobase_read_doc,
-    "Read at most size characters from stream.\n"
-    "\n"
-    "Read from underlying buffer until we have size characters or we hit EOF.\n"
-    "If size is negative or omitted, read until EOF.\n"
-    );
+/*[clinic input]
+_io._TextIOBase.read
+    cls: defining_class
+    /
+    *args: object
```

Comment:
IMO the signature is wrong: it should be 0 or 1 positional argument, or maybe 1 optional "size" keyword argument.

Reply:
Yes, I'm following up all these in a single PR.